src/microbe_stage/systems/StrainSystem.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace Systems
+{
+    using Components;
+    using DefaultEcs;
+    using DefaultEcs.System;
+    using DefaultEcs.Threading;
+    using Godot;
+    using World = DefaultEcs.World;
+
+    [With(typeof(CompoundStorage))]
+    [With(typeof(OrganelleContainer))]
+    [With(typeof(MicrobeControl))]
+    [With(typeof(StrainAffected))]
+    [ReadsComponent(typeof(OrganelleContainer))]
+    [ReadsComponent(typeof(MicrobeControl))]
+    public sealed class StrainSystem : AEntitySetSystem<float>
+    {
+        public StrainSystem(World world, IParallelRunner runner) : base(world, runner)
+        {
+        }
+
+        protected override void Update(float delta, in Entity entity)
+        {
+            ref var strain = ref entity.Get<StrainAffected>();
+            ref var organelles = ref entity.Get<OrganelleContainer>();
+
+            if (strain.IsUnderStrain)
+            {
+                var strainIncrease = Constants.SPRINTING_STRAIN_INCREASE_PER_SECOND * delta;
+                strainIncrease += organelles.HexCount * Constants.SPRINTING_STRAIN_INCREASE_PER_HEX;
+
+                if (strain.CurrentStrain > Constants.MAX_STRAIN_PER_CELL)
```

Comment:
Shouldn't this check be below line 35 so that the new value is correctly clamped?